phoenix-core/src/it/java/org/apache/phoenix/end2end/RowValueConstructorOffsetIT.java

```diff
@@ -0,0 +1,943 @@
+                        assertEquals(1, result);
+                        nRows++;
+                    }
+                }
+            }
+            conn.commit();
+        }
+
+        String createIndex = "CREATE INDEX IF NOT EXISTS " + INDEX_NAME + " ON " + TABLE_NAME + " (k2 DESC,k1)";
+        try (Statement statement = conn.createStatement()) {
+            statement.execute(createIndex);
+        }
+
+        String createDataIndex = "CREATE INDEX IF NOT EXISTS " + DATA_INDEX_NAME + " ON " + DATA_TABLE_NAME
+                + " (k2 DESC,k1)";
+        try (Statement statement = conn.createStatement()) {
+            statement.execute(createDataIndex);
+        }
+        conn.commit();
+    }
+
+    @AfterClass
+    public static void cleanup() {
+        try {
+            if (conn != null) {
+                conn.close();
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    // Test RVC Offset columns must be coercible to a base table
+    @Test
+    public void testRVCOffsetNotCoercible() throws SQLException {
+        //'ab' is not an integer so this fails
+        String failureSql = String.format("SELECT %s FROM %s OFFSET (%s)=('a', 'ab', 2)",
+                TABLE_ROW_KEY,TABLE_NAME,TABLE_ROW_KEY);
+        try (Statement statement = conn.createStatement()){
+            statement.execute(failureSql);
+            fail("Should not allow non coercible values to PK in RVC Offset");
+        } catch (RowValueConstructorOffsetNotCoercibleException e) {
+            return;
+        }
+    }
+
+    // Test Order By Not PK Order By Exception
+    @Test
+    public void testRVCOffsetNotAllowNonPKOrderBy() throws SQLException {
+        String failureSql = String.format("SELECT %s, v1 FROM %s ORDER BY v1 OFFSET (%s)=(%s)",
+                TABLE_ROW_KEY,TABLE_NAME,TABLE_ROW_KEY,GOOD_TABLE_ROW_KEY_VALUE);
+        try (Statement statement = conn.createStatement()) {
+            statement.execute(failureSql);
+            fail("Should not allow no PK order by with RVC Offset");
+        } catch (RowValueConstructorOffsetNotCoercibleException e) {
+            return;
+        }
+
+    }
+
+    // Test Order By Partial PK Order By Exception
+    @Test
+    public void testRVCOffsetNotAllowPartialPKOrderBy() throws SQLException {
+        String failureSql = String.format("SELECT %s FROM %s ORDER BY k1 OFFSET (%s)=(%s)",
+                TABLE_ROW_KEY,TABLE_NAME, TABLE_ROW_KEY, GOOD_TABLE_ROW_KEY_VALUE);
+        try (Statement statement = conn.createStatement()){
+            statement.execute(failureSql);
+            fail("Should not allow partial PK order by with RVC Offset");
+        } catch (RowValueConstructorOffsetNotCoercibleException e) {
+            return;
+        }
+    }
+
+    // Test Order By Different Sort PK Order By Exception
+    @Test
+    public void testRVCOffsetSamePKDifferentSortOrderBy() throws SQLException {
```

Comment:
Just wondering whether it is necessary to also have a test where the table itself has a mix of descending/ascending pk. The current test has all ascending pk. I'm open to excluding such a test if you think it is not required @dbwong

Reply:
I think this test is not required for RowValueConstructorOffsetIT as ascending/descending is not a consideration.  However, the QueryMoreIT where this is a important case that we have the test testRVCWithDescAndAscendingPK which covers this.